src/asset/asset.js

```diff
@@ -189,6 +209,49 @@ Object.assign(pc, function () {
             return this.file;
         },
 
+        /**
+         * @private
+         * @function
+         * @name pc.Asset#getLocalizedAssetId
+         * @param {String} locale The desired locale e.g. ar-AR.
+         * @description Returns the asset id of the asset that corresponds to the specified locale.
+         * @returns {Number} An asset id or null if there is no asset specified for the desired locale.
+         */
+        getLocalizedAssetId: function (locale) {
+            // tries to find either the desired locale or a fallback locale
+            locale = pc.I18n.findAvailableLocale(locale, this._i18n);
+            return this._i18n[locale] || null;
+        },
+
+        /**
+         * @private
+         * @function
+         * @name pc.Asset#setLocalizedAssetId
+         * @param {String} locale The locale e.g. ar-AR.
+         * @param {Number} assetId The asset id
+         * @description Sets a replacement asset id for the specified locale. When the locale in {@link pc.Application#i18n} changes then
+         * references to this asset will be replaced with the specified asset id. (Currently only supported by the {@link pc.ElementComponent}).
+         */
+        setLocalizedAssetId: function (locale, assetId) {
```

Comment:
Same comment as the event. Should we pair add/remove rather than set/remove

Reply:
We could do that sure.